Option to add podAnnotations to jobs.

novotl opened the pull request:
Adds an option to add `podAnnotations` to jobs. The style is the same as with `deployments` e.g. here https://github.com/acryldata/datahub-helm/blob/21d4f781d7b0cb276a3e6bf076ebeaafabc5b48c/charts/datahub/subcharts/datahub-gms/templates/deployment.yaml#L12-L17

Motivation:
- adding [vault annotations](https://banzaicloud.com/docs/bank-vaults/mutating-webhook/annotations/) to inject secrets from vault - many jobs require some sort of connection string, that may be usually stored in vault
- adding [datadog annotations](https://docs.datadoghq.com/agent/kubernetes/integrations/?tab=kubernetes#configuration) to configure log collection

## Checklist
- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [x] Links to related issues (if applicable)
- [x] Tests for the changes have been added/updated (if applicable)
  - Tested my changes locally with `helm template charts/datahub -f charts/datahub/values.yaml`
  - Tried adding annotations to `charts/datahub/values.yaml` and rendering templates again
- [ ] Docs related to the changes have been added/updated (if applicable)

tengis commented:
Plus +1 here, please. This one is a much-needed feature! Thanks, Team!

tengis commented:
Hi @novotl, would you mind fixing the linting here? Thanks

pedro93 commented:
Hello @novotl 

Overall looks good to me, thank you for making this PR!
Just have 2 minor points if you could quickly address them:
1.) Include `datahub-ingestion-cron` in this PR.
2.) For all resources that were changed, please update the `README.md` & `values.yaml` files with documentation & comments respectively on how this feature can be used. You can check the existing information in those files to understand how we do it. If it is unclear feel free to ping me on Slack or here!

Thanks again for your contribution!!

novotl commented:
Thanks for the feedback @pedro93, addressed your comments in:
1) cron - https://github.com/acryldata/datahub-helm/pull/137/commits/3e966dd75b84ee27ce0a650d20d910bab55fcc31
2) documentation - https://github.com/acryldata/datahub-helm/pull/137/commits/1a1d5e47b41608cf72d09eed1ceb6bfaf087659f

are those ok?

pedro93 commented:
Please bump the chart.yaml version, other than that looks good to me!

novotl commented:
> Please bump the chart.yaml version, other than that looks good to me!

Sure 👉 https://github.com/acryldata/datahub-helm/pull/137/commits/f0ede6cf6449a8e9f3898a50f255b15d54522074
